src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -309,24 +310,44 @@ public Integer replaceEntityProperties(
     @Timed
     @PutMapping( value = "/" + ASSOCIATION, consumes = MediaType.APPLICATION_JSON_VALUE )
     public Integer createAssociations( @RequestBody Set<DataEdgeKey> associations ) {
+        final var srcAssociationEntitySetIds = new HashMap<UUID, Set<UUID>>(); // edge-src
+        final var dstAssociationEntitySetIds = new HashMap<UUID, Set<UUID>>(); // edge-dst
+
+        final var entitySetIdChecks = new HashMap<AclKey, EnumSet<Permission>>();
+        final SetMultimap<UUID, UUID> keyPropertiesOfAssociations = HashMultimap.create();
         associations.forEach(
                 association -> {
-                    UUID associationEntitySetId = association.getEdge().getEntitySetId();
-
-                    //Ensure that we have read access to entity set metadata.
-                    ensureReadAccess( new AclKey( association.getSrc().getEntitySetId() ) );
-                    ensureReadAccess( new AclKey( association.getDst().getEntitySetId() ) );
-                    ensureReadAccess( new AclKey( associationEntitySetId ) );
-
-                    //Ensure that we can read key properties.
-                    Set<UUID> keyPropertyTypes = edmService
-                            .getEntityTypeByEntitySetId( associationEntitySetId ).getKey();
-                    keyPropertyTypes.forEach( propertyType ->
-                            accessCheck( new AclKey( associationEntitySetId, propertyType ), READ_PERMISSION ) );
+                    final var edgeEntitySetId = association.getEdge().getEntitySetId();
+                    final var srcEntitySetId = association.getSrc().getEntitySetId();
+                    final var dstEntitySetId = association.getDst().getEntitySetId();
+
+                    entitySetIdChecks.put( new AclKey( edgeEntitySetId ), READ_PERMISSION );
+                    entitySetIdChecks.put( new AclKey( srcEntitySetId ), READ_PERMISSION );
+                    entitySetIdChecks.put( new AclKey( dstEntitySetId ), READ_PERMISSION );
```

Comment:
Should these be write permissions?

Reply:
This call does not write any data, so I guess that's why you don't need write permission. The original checks were implemented here: https://jira.openlattice.com/browse/LATTICE-1738